src/benchmarks/gc/docs/suites.md

```diff
@@ -0,0 +1,127 @@
+# Suites
+
+Apart from individual tests, GC Benchmarking Infra can run and process results
+in sets with its suite functionality. This is specified in a `suite.yaml` file,
+which has the following structure:
+
+```yml
+bench_files:
+- test1.yaml
+- test2.yaml
+- test3.yaml
+- test4.yaml
+command_groups:
+  example_command_group:
+  - command args
+  - command args
+  - command args
+```
+
+These sections are explained with examples in the following sections.
+
+## Suite Run
+
+The `suite-run` command allows you to run a set of tests by specifying the `yaml`
+files in the order you want them to be run. GC Infra will read and parse each one
+of these files and run the benchmark tests accordingly. If for some reason an
+error occurs, that test will be ignored and infra will proceed to the next one.
+
+At the end of the run, if any problems were encountered, a summary detailing the
+exceptions, the respective messages, and the stack traces will be shown to the user.
+
+You can write your own suite files to tailor to your specific performance benchmarking
+needs. Let's take the suite generated by default as an example.
+
+```yml
+bench_files:
+- normal_workstation.yaml
+- normal_server.yaml
+- high_memory.yaml
+- low_memory_container.yaml
+command_groups: {}
+```
+
+We would run this by issuing the following command:
+
+```sh
+py . suite-run bench/suite/suite.yaml
+```
+
+The result of this is the equivalent of individually running each test like this:
+
+```sh
+py . run bench/suite/normal_workstation.yaml
+py . run bench/suite/normal_server.yaml
+py . run bench/suite/high_memory.yaml
+py . run bench/suite/low_memory_container.yaml
+```
+
+Once the last test has finished running, if everything went fine, you'll be
+greeted with a message saying "*\*\*\* Suite run finished successfully! \*\*\**".
+
+If something went wrong somewhere, you will be shown a summary with the following
+format:
+
+```text
+*WARNING*: One or more tests in the suite encountered errors.
+
+*** Here is a summary of the problems found: ***
+
+======= <Yaml Filename> =======
+
+===== <Coreclr Name> =====
+
+=== <Configuration Name> ===
+
+- <Benchmark Name> -
+<Iteration Number>
+<Error Message>
+
+<Stack Trace>
+
+<Repeat with other errors that might have occurred>
+```
+
+## Suite Run Command
```

Comment:
interesting, I had thought that we just ran the command_groups after running each scenario 'cause otherwise it seems a little odd to have it in the same .yaml if it's a complete different thing. is there any relation between command_groups and the rest in suite.yaml?

Reply:
I thought the `command_groups` were part of the `suite-run` as such, but after looking into the source code and running various experiments, I learned they are completely different. Maybe they were added here as to bundle all suite-like stuff in one file?

Reply:
that seems like a weird design... I don't have a better explanation than what you have.